src/common.rs

```diff
@@ -66,7 +74,7 @@ pub(crate) async fn new_grpc_channel(
 /// Returns google token (String value) from
 /// Google Cloud Platform project JSON credentials (provided as String).
 #[allow(clippy::rc_buffer)]
-pub(crate) fn get_token(google_credentials: impl AsRef<str>) -> Result<Arc<String>> {
+pub(crate) fn get_token(google_credentials: impl AsRef<str>) -> CognativeResult<Arc<String>> {
```

Comment:
I would really like to stick with Result

Reply:
Would you be okay with `errors::Result`? Interceptor needs to have a return type which uses Rust's inbuilt result type that takes the error generic. Otherwise the interceptor can live in `inteceptor.rs` or something maybe?

Reply:
let''s create alias for core::result::Result instead, e.g. **StdResult**

This way there is not need to change our Result on multiple places, just to use StdResult alias in interceptor definition. Also as said earlier CognativeResult contains typo, should be Cogn**i**tiveResult. My preference is to use StdResult for standard result here.